dipy/tracking/life.py

```diff
@@ -306,9 +307,7 @@ def voxel2streamline(streamline, transformed=False, affine=None,
 
     if unique_idx is None:
         all_coords = np.concatenate(transformed_streamline)
-        unique_idx = unique_rows(all_coords.astype(int))
-    else:
-        unique_idx = unique_idx
+        unique_idx = unique_rows(np.round(all_coords).astype(np.intp))
```

Comment:
Also - need test for modified behavior - such as `transformed_streamline` with non-integer indices.